app/cluster/dynamic.go

```diff
@@ -108,15 +108,17 @@ func (d *Dynamic) Run(ctx context.Context) error {
 			}
 
 			d.logger.Infof("Got trigger to change the mode, new mode: %s, old mode: %s", req.Mode(), d.currentMode)
-			err := d.handleModeChange(req.Mode())
-			if err != nil {
-				d.logger.Error(err)
-				return err
-			}
-			d.logger.Debugf("Acknowledging the mode change")
-
-			if err := req.Ack(ctx); err != nil {
-				return fmt.Errorf("ack mode change: %w", err)
+			if !d.GatewayComponent {
```

Comment:
Wouldn't it be easier to follow if we just `continue` if it's a gateway? Like:

```go
if d.GatewayComponent {
    continue
}

// do the rest
```

Reply:
Yeah this makes sense...